dashboard/test/controllers/api_controller_test.rb

```diff
@@ -1600,4 +1600,30 @@ class ApiControllerTest < ActionController::TestCase
       {method: "get", path: "/api/student_progress/2/15"}
     )
   end
+
+  def assert_levelgroup_results_match(expected_results, actual_results)
+    # Results may be in any order, so...
+    # Assert sets of equal length
+    assert_equal expected_results.count, actual_results.count
+    # Assert every expected result is found in the actual results
+    expected_results.each do |expected_result|
```

Comment:
The previous code did a second check, flipping the variables.  I think this protected against duplicate entries which would pad out the length to match but not actually be a proper match of results?

Reply:
My thinking was:
- Given we assert the two sets are of equal size
- Then asserting all A is found in B is equivalent to asserting all B is found in A.
Now that you bring it up though, I realize this isn't true if the elements in A and B are not unique (possible since they are lists, not actually sets).  Basically `[A A B B B], [A A A B B]` would pass this test when it should not.  Unfortunately, asserting all B is found in A also wouldn't be sufficient to catch that case, so we probably need to remove elements as they're matched to do this right.

Reply:
well-reasoned!